ioctl/cmd/action/stake2register.go

```diff
@@ -0,0 +1,122 @@
+	}
+
+	registerCmdShorts = map[config.Language]string{
+		config.English: "register a candidate",
+		config.Chinese: "注册候选人",
+	}
+
+	stake2FlagRegisterAutoRestakeUsages = map[config.Language]string{
+		config.English: "auto restake without power decay",
+		config.Chinese: "自动质押，权重不会衰减",
+	}
+)
+
+// stake2RegisterCmd represents the stake2 register a candidate command
+var stake2RegisterCmd = &cobra.Command{
+	Use:   config.TranslateInLang(registerCmdUses, config.UILanguage),
+	Short: config.TranslateInLang(registerCmdShorts, config.UILanguage),
+	Args:  cobra.RangeArgs(6, 7),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := register(args)
+		return output.PrintError(err)
+	},
+}
+
+func init() {
+	registerWriteCommand(stake2RegisterCmd)
+	stake2RegisterCmd.Flags().BoolVar(&stake2AutoRestake, "auto-restake", false, config.TranslateInLang(stake2FlagRegisterAutoRestakeUsages, config.UILanguage))
+}
+
+func register(args []string) error {
+	name := args[0]
+	if err := validator.ValidateCandidateNameForStake2(name); err != nil {
+		return output.NewError(output.ValidationError, "invalid candidate name", err)
+	}
+
+	operatorAddrStr, err := util.Address(args[1])
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to get operator address", err)
+	}
+	rewardAddrStr, err := util.Address(args[2])
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to get reward address", err)
+	}
+	ownerAddrStr, err := util.Address(args[3])
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to get owner address", err)
+	}
+
+	amountStr := args[4]
+	amountInRau, err := util.StringToRau(amountStr, util.IotxDecimalNum)
+	if err != nil {
+		return output.NewError(output.ConvertError, "invalid amount", err)
+	}
+	amountStringInRau := amountInRau.String()
+
+	stakeDuration, err := parseStakeDuration(args[5])
+	if err != nil {
+		return output.NewError(0, "", err)
+	}
+	duration := uint32(stakeDuration.Uint64())
+
+	var payload []byte
+	if len(args) == 7 {
+		payload = make([]byte, 2*len([]byte(args[6])))
+		hex.Encode(payload, []byte(args[6]))
+	}
+
+	sender, err := signer()
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to get signed address", err)
+	}
+
+	gasLimit := gasLimitFlag.Value().(uint64)
+	if gasLimit == 0 {
+		gasLimit = action.CandidateRegisterBaseIntrinsicGas +
+			action.CreateStakePayloadGas*uint64(len(payload))
+	}
+
+	gasPriceRau, err := gasPriceInRau()
+	if err != nil {
+		return output.NewError(0, "failed to get gas price", err)
+	}
+	nonce, err := nonce(sender)
+	if err != nil {
+		return output.NewError(0, "failed to get nonce ", err)
+	}
+	cr, err := action.NewCandidateRegister(nonce, name, operatorAddrStr, rewardAddrStr, ownerAddrStr, amountStringInRau, duration, autoRestake, payload, gasLimit, gasPriceRau)
```

Comment:
can directly use amountInRau.String() here

Reply:
need to use stake2AutoRestake here, right?

Reply:
sure

Reply:
other PRs don't have same issue.